### PR TITLE
Add Unit Test GitHub Actions Workflow

### PR DESCRIPTION
**Pull request by kpriver55**
## WHAT

Added a GitHub Actions workflow that will automatically run vectorflow's unit tests when a pull request is opened, re-opened or edited.

## VERIFICATION
Pictures below show that, with no changes to the core code, the jobs build and complete, passing all checks.

Screenshot 2023-10-27 204010

Screenshot 2023-10-27 204029

Screenshot 2023-10-27 204044

EDIT:
Additional screenshots showing the the unit tests pass:
API:

Screenshot 2023-10-30 092539

Worker:

Screenshot 2023-10-30 092633

## CAUTION
In order for this to work properly, environment variables must be set through GitHub Secrets. You will need to set the LOCAL_VECTOR_DB variable in order for the tests to run. It shouldn't matter what the value of the variable is, as the reason the variable needs to be set is that there's a line in the API unit test code that deletes this variable leading to an error if the variable doesn't exist.

@dgarnitz

### CODE REVIEW

**Comment by dgarnitz** [comment]
@kpriver55  Please document in this PR all the environment variables we need to set in the main repo to get this to work. 

Can you also expand the sections in the screenshots like `Run api unit tests` so that the verification evidence proves that the tests pass. With github actions its possible for an action run to complete and the tests to not run at all but because it does not throw an outright failure, the actions runner still interprets it as a successfully run. Please re-run and add new screenshots

**Review by dgarnitz — COMMENTED** [review]
This looks great! just add the new Verification evidence and info about env vars to set and it will be good to go

**Comment by dgarnitz** [comment]
@kpriver55 if it does not matter what value it is set to, please set it to something random like `qdrant`.

File: .github/workflows/run_unittests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10"]
+
+    env:
+      LOCAL_VECTOR_DB: ${{ secrets.LOCAL_VECTOR_DB }}
```
Comment (dgarnitz) [inline .github/workflows/run_unittests.yml:19]: set this equal to `qdrant` if the value does not matter. let's not use secrets where they are not needed.

**Review by dgarnitz — CHANGES_REQUESTED** [review]

File: .github/workflows/run_unittests.yml (hunk 1 above)
Comment (kpriver55) [reply .github/workflows/run_unittests.yml]: I've changed the value. Everything still works @dgarnitz

**Review by dgarnitz — APPROVED** [review]

File: .github/workflows/run_unittests.yml (hunk 1 above)
Comment (dgarnitz) [reply .github/workflows/run_unittests.yml]: Awesome. thank you!

**Pull request merged**